etl/steps/data/garden/oecd/2023-05-19/co2_air_transport.py

```diff
@@ -0,0 +1,307 @@
+    log.info("co2_air_transport.start")
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow: Dataset = paths.load_dependency("co2_air_transport")
+    ds_tour: Dataset = paths.load_dependency("unwto")
+
+    # Read table from meadow dataset.
+    tb_meadow = ds_meadow["co2_air_transport"]
+    tb_tour = ds_tour["unwto"]
+    df = pd.DataFrame(tb_meadow)  # Create a dataframe with data from the co2 transport table.
+    df_tr = pd.DataFrame(tb_tour)  # Create a dataframe with data from the tourism table.
+
+    # list of months
+    month_names = [
+        "January",
+        "February",
+        "March",
+        "April",
+        "May",
+        "June",
+        "July",
+        "August",
+        "September",
+        "October",
+        "November",
+        "December",
+    ]
+
+    #
+    # Process data.
+    #
+    log.info("co2_air_transport.harmonize_countries")
+
+    df = geo.harmonize_countries(df=df, countries_file=paths.country_mapping_path)
+    # Only use passenger flights and total international/domestic emissions
+    df = df[df["flight_type"] == "P"]
+    df.drop("flight_type", axis=1, inplace=True)
+    df = df[df["emission_source"].isin(["TER_DOM", "TER_INT"])]
+    df = df.reset_index(drop=True)
+
+    # Process annual data
+    pivot_table_ye = process_annual_data(df)
+
+    # Add population data to the DataFrame
+    pivot_table_ye = geo.add_population_to_dataframe(
+        pivot_table_ye, country_col="country", year_col="year", population_col="population"
+    )
+
+    emissions_columns = pivot_table_ye.columns[2:-1]
+
+    # Generate per capital co2 emissions data and add it do the dataframe
+    for col in emissions_columns:
+        pivot_table_ye[f"per_capita_{col}"] = pivot_table_ye[col] / pivot_table_ye["population"]
+
+    # Add Inbound/Outbound tourism to the dataframe (multiply international aviation emissions by international arrivals/departures)
+    pivot_outb = add_inbound_outbound_tour(pivot_table_ye, df_tr)
+
+    # Process monthly data
+    pivot_df, pivot_table_mn = process_monthly_data(df, month_names)
+
+    concatenated_df = pd.merge(pivot_outb, pivot_table_mn, on=["year", "country"], how="outer")
+    merge_df = pd.merge(concatenated_df, pivot_df, on=["year", "country"], how="outer")
+
+    merge_df = ukraine_fill_war_for_reg_agg(merge_df)
+    merge_df = merge_df.drop("TER_INT_m", axis=1)
+
+    merge_df = merge_df.rename(columns={"ter_int_m_filled_ukraine": "TER_INT_m"})
+
+    regions_ = ["North America", "South America", "Europe", "Africa", "Asia", "Oceania", "European Union (27)"]
+
+    for region in regions_:
+        merge_df = geo.add_region_aggregates(df=merge_df, country_col="country", year_col="year", region=region)
+
+    merge_df["TER_INT_m"] = merge_df["TER_INT_m"].replace(0, np.nan)
+    merge_df = merge_df[merge_df["year"] != 2023]
+
+    # Apply the function to each row using apply()
+    merge_df = merge_df.apply(update_values, axis=1)
```

Comment:
The entire `apply` could be simplified by this
```
merge_df.loc[~merge_df.country.isin(regions_), month_names] = np.nan
```
(the biggest benefit is that it's much faster)